File: kelly/스택 & 큐/두 큐 합 같게 만들기/Solution02.java
class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int qSize = queue1.length;
        int[] dq = new int[qSize * 2];
        long q1Sum = 0;
        long q2Sum = 0;

        for (int i = 0; i < qSize; i++) {
            dq[i] = queue1[i];
            q1Sum += queue1[i];
        }

        for (int i = 0; i < qSize; i++) {
            dq[qSize + i] = queue2[i];
            q2Sum += queue2[i];
        }

        int answer = 0;
        int left = 0;
        int right = qSize;

        for (int i = 0; i < qSize * 4; i++) {
            if (q1Sum == q2Sum) {
                return answer;
            }

            if (q1Sum > q2Sum) {
                q1Sum -= dq[left];
                q2Sum += dq[left];
                left = (left + 1) % (dq.length);
            } else {
                q1Sum += dq[right];
                q2Sum -= dq[right];
                right = (right + 1) % (dq.length);
            }

            answer += 1;
        }

        return -1;
    }
}

File: kelly/스택 & 큐 & PQ/두 큐 합 같게 만들기/solution02.py
def solution(queue1, queue2):
    q_size = len(queue1)
    q1_sum = sum(queue1)
    q2_sum = sum(queue2)

    dq = queue1[::]
    dq.extend(queue2)

    answer = 0
    left = 0
    right = q_size

    for i in range(q_size * 4):
        if q1_sum == q2_sum:
            return answer

        if q1_sum > q2_sum:
            q1_sum -= dq[left]
            q2_sum += dq[left]
            left = (left + 1) % len(dq)
        else:
            q1_sum += dq[right]
            q2_sum -= dq[right]
            right = (right + 1) % len(dq)

        answer += 1

    return -1